pysam/libcalignmentfile.pyx

```diff
@@ -1766,7 +1761,8 @@ cdef class IteratorRow:
             # better: use seek/tell?
             with nogil:
                 hdr = sam_hdr_read(self.htsfile)
-            assert hdr != NULL
+            if hdr is NULL:
```

Comment:
I'm not sure how Cython treats tests for object identity with `NULL`.  I typically use a boolean test instead: `if not hdr`.